frontend/packages/dev-console/src/actions/modify-application.ts

```diff
@@ -25,8 +26,13 @@ export const ModifyApplication = (kind: K8sKind, obj: K8sResourceKind): KebabOpt
 };
 
 export const EditApplication = (model: K8sKind, obj: K8sResourceKind): KebabOption => {
+  const annotation =
+    obj.metadata.annotations && obj.metadata.annotations['openshift.io/generated-by'];
   return {
     label: `Edit ${truncateMiddle(obj.metadata.name, { length: RESOURCE_NAME_TRUNCATE_LENGTH })}`,
+    hidden:
+      (obj.kind !== ServiceModel.kind && !annotation) ||
+      (annotation && annotation !== 'OpenShiftWebConsole'),
```

Comment:
I'd prefer if we had 2 actions: `EditApplication` and `EditWebConsoleGeneratedApplication` (naming is hard...)

The first is the current action with no `hidden` property and it will be used by knative service.
The second will be used by DC and D models.

Note that a further change is needed still in `dev-console/src/utils/kebab-actions` to ensure we only contribute this action to D and DC. Currently it's also present for DS and SS.

Reply:
I will add 2 separate actions ```EditApplication``` for knative resources and ```EditWebConsoleGeneratedApplication``` for DC and D. And no actions DS and SS.